scripts/mirror_lkl_layer.py

```diff
@@ -0,0 +1,131 @@
+            return field['name']
+        
+        
+def main(arguments):
+    # initialize logger
+    logger = initialize_logging(arguments.log_file)
+    # Create the GIS
+    logger.info("Authenticating...")
+    # First step is to get authenticate and get a valid token
+    gis = GIS(arguments.org_url,
+              username=arguments.username,
+              password=arguments.password,
+              verify_cert=not arguments.skip_ssl_verification)
+    
+    # Get the feature layer
+    if gis.content.get(arguments.item_id):
+        logger.info("Getting feature layer")
+        item = gis.content.get(arguments.item_id)
+        mirror_layer = item.layers[0]
+        if arguments.lkl_layer_url:
+            lkl_layer = FeatureLayer(url=arguments.lkl_layer_url)
+        else:
+            lkl_layer = gis.admin.location_tracking.last_known_locations_layer
+            
+        # Query LKL and mirror layer
+        lkl_fset = lkl_layer.query('1=1', out_sr=3857)
+        if len(lkl_fset) == 0:
+            logger.info("No LKLs in your layer yet!")
+            sys.exit(0)
+        mirror_fset = mirror_layer.query('1=1', out_sr=3857)
+        
+        add_features = []
+        update_features = []
+        logger.info("Iterating through current LKL data")
+        for feature in lkl_fset:
+            found = False
+            for mirror_feature in mirror_fset:
+                # use "in" instead of == comparison due to the potential for brackets to be in the GUID field
+                if mirror_feature.attributes[return_field_name(mirror_layer, "global_id")].lower() in feature.attributes["globalid"].lower():
+                    found = True
+                    break
+            if found:
+                update_features.append(feature)
+            else:
+                add_features.append(feature)
+                
+        logger.info("Posting updated data to mirrored layer")
+        mirror_layer.edit_features(adds=add_features, updates=update_features, use_global_ids=True)
+        logger.info("Completed!")
+    else:
+        logger.info("Item not found")
+        
+
+if __name__ == "__main__":
+    # Get all of the commandline arguments
+    parser = argparse.ArgumentParser("Python script which maintains an exact replica of your LKL layer in a separate feature service, so that data can be joined")
+    parser.add_argument('-u', dest='username', help="The username to authenticate with", required=True)
+    parser.add_argument('-p', dest='password', help="The password to authenticate with", required=True)
+    parser.add_argument('-org', dest='org_url', help="The url of the org/portal to use", required=True)
+    # Parameters for tracker
+    parser.add_argument('-item-id', dest='item_id', required=True,
+                        help="The item id of the layer you want the tracks to mirror to")
+    parser.add_argument('-lkl-layer-url', dest='lkl_layer_url', default=None,
```

Comment:
I would just make this a required parameter and not bother with the admin access. I think most of the time it will be a specific view that will be mirrored.